File: dp-web/src/app/services/configuration.service.ts
/*
 *
 *  * Copyright  (c) 2016-2017, Hortonworks Inc.  All rights reserved.
 *  *
 *  * Except as expressly permitted in a written agreement between you or your company
 *  * and Hortonworks, Inc. or an authorized affiliate or partner thereof, any use,
 *  * reproduction, modification, redistribution, sharing, lending or other exploitation
 *  * of all or any part of the contents of this software is strictly prohibited.
 *
 */

import { Injectable } from '@angular/core';
import { Http, RequestOptions } from '@angular/http';
import { Observable } from 'rxjs/Observable';

import { User } from '../models/user';

import { HttpUtil } from '../shared/utils/httpUtil';
import {LDAPProperties, LDAPUpdateProperties} from '../models/ldap-properties';

@Injectable()
export class ConfigurationService {
  uri = 'api/init';
  knowConfigUri = 'api/knox';
  configUri = 'api/config';

  constructor(private http:Http) { }

  retrieve(): Observable<{
    user: User,
    lakeWasInitialized: boolean,
    authWasInitialized: boolean,
    rbacWasInitialized: boolean
  }>{
    return this.http
      .get(this.uri, new RequestOptions(HttpUtil.getHeaders()))
      .map(HttpUtil.extractData)
      .catch(HttpUtil.handleError);
  }

  configureLDAP(ldapProperties: LDAPProperties): Observable<LDAPProperties>{
    return this.http
      .post(`${this.knowConfigUri}/configure`,ldapProperties, new RequestOptions(HttpUtil.getHeaders()))
      .map(HttpUtil.extractData)
      .catch(HttpUtil.handleError)
  }

  updateLDAP(ldapUpdateProperties: LDAPUpdateProperties): Observable<LDAPUpdateProperties>{
    return this.http
      .post(`${this.knowConfigUri}/ldap`,ldapUpdateProperties, new RequestOptions(HttpUtil.getHeaders()))
      .map(HttpUtil.extractData)
      .catch(HttpUtil.handleError)
  }

  isKnoxConfigured(): Observable<any> {
    return this.http
      .get(`${this.knowConfigUri}/status`, new RequestOptions(HttpUtil.getHeaders()))
      .map(HttpUtil.extractData)
      .catch(HttpUtil.handleError);
  }

  getLdapConfiguration(): Observable<LDAPProperties> {
    return this.http
      .get(`${this.knowConfigUri}/ldap`, new RequestOptions(HttpUtil.getHeaders()))
      .map(HttpUtil.extractData)
      .catch(HttpUtil.handleError);
  }

  getGATrackingStatus() : Observable<boolean>{
    return this.http.get(`${this.configUri}/ga-tracking-status`, new RequestOptions(HttpUtil.getHeaders()))
      .map((response:any) => response._body === "true")
      .catch(HttpUtil.handleError);
  }
}

File: services/gateway/src/main/java/com/hortonworks/dataplane/gateway/service/ConfigurationService.java
/*
 *
 *  * Copyright  (c) 2016-2017, Hortonworks Inc.  All rights reserved.
 *  *
 *  * Except as expressly permitted in a written agreement between you or your company
 *  * and Hortonworks, Inc. or an authorized affiliate or partner thereof, any use,
 *  * reproduction, modification, redistribution, sharing, lending or other exploitation
 *  * of all or any part of the contents of this software is strictly prohibited.
 *
 */
package com.hortonworks.dataplane.gateway.service;

import com.hortonworks.dataplane.gateway.domain.KnoxConfigurationResponse;
import feign.FeignException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class ConfigurationService {
  private static final Logger logger = LoggerFactory.getLogger(ConfigurationService.class);
  @Autowired
  private ConfigurationServiceInterface configurationServiceInterface;

  private static final String DP_SESSION_TIMEOUT_KEY = "dp.session.timeout.minutes";

  public boolean isLdapConfigured() {
    try {
      KnoxConfigurationResponse knoxStatus = configurationServiceInterface.getKnoxStatus();
      return knoxStatus.getConfigured();
    } catch (FeignException e) {
      logger.error("error while calling configuration service", e);
      throw new RuntimeException(e);//TODO should we just sned false?
    }
  }


  public Long getJwtTokenValidity() {
    try {
      String validityStr = configurationServiceInterface.getTokenValidity(DP_SESSION_TIMEOUT_KEY);
      if (validityStr.isEmpty()) {
        return -1L;
      }
      return Long.parseLong(validityStr);
    } catch (FeignException e) {
      logger.error("error while calling configuration service", e);
      throw new RuntimeException(e);
    }
  }
}
